Formula/swift-sh.rb

```diff
@@ -0,0 +1,19 @@
+class SwiftSh < Formula
+  desc "Scripting with easy zero-conf dependency imports"
+  homepage "https://github.com/mxcl/swift-sh"
+  url "https://github.com/mxcl/swift-sh/archive/1.16.1.tar.gz"
+  sha256 "fb70b075101594b146d4120b52d6132f8b34d3337d4d55ed289c57e4d7808519"
+  head "https://github.com/mxcl/swift-sh.git"
+
+  depends_on :xcode => ["10.0", :build]
+
+  def install
+    system "swift", "build", "--disable-sandbox", "-c", "release"
+    bin.install ".build/release/swift-sh"
+    bin.install ".build/release/swift-sh-edit"
+  end
+
+  test do
+    assert_match "swift sh", shell_output("#{bin}/swift-sh --help")
```

Comment:
You're almost there but we really need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
As I have mentioned upwards, I first wrote a test checking the basic functionality (writing a script and checks if it compiles right), but unfortunately it tries to write outside the sandbox (it tries to access the Library directory) and fails. Is there any method to run the test without the sandbox?

Reply:
No there isn't, this is something you'll have to work with upstream for to resolve

Reply:
@SMillerDev Looks like other formulas such as [Sourcery](https://github.com/Homebrew/homebrew-core/blob/master/Formula/sourcery.rb) that writes cache files [also just checks the `--version`](https://github.com/Homebrew/homebrew-core/blob/d6e1f35b46fc4aa864ba1273af3f7e79fbaa1145/Formula/sourcery.rb).

Reply:
Yes, some formula were accepted before we started enforcing this requirement. It has improved the situation for both maintainers and users so we're not going to stray from it.